### PR TITLE
Global User Recommendation Feature Switch

### PR DESCRIPTION
**Pull request by tailorck**
## Description

This is a new feature to allow for a global boolean to enable/disable all user recommendations

## Motivation and Context

We need a quick way to "turn off" parqr without disturbing data collection and admin dashboard

## How Has This Been Tested?

It hasn't

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have run the current suite of unit tests to make sure my code doesn't break anything

### CODE REVIEW

**Review by RFinkelberg — COMMENTED** [review]
LGTM but a few considerations:
1. Do we need to do this on a per class basis? Essentially support config files for each class with switches like these?
2. Will this affect the dashboard too? 
3. How is the frontend handling this change? Are we showing nothing, or will the UI look broken? We may need to send back a special response code if the switch is off, so the frontend knows what to do

File: app/utils.py [hunk 1]
```diff
@@ -25,3 +26,15 @@ def wrapper(*args, **kwargs):
...
+
+
+def global_user_recommendation_switch(func):
+
+    @wraps(func)
+    def wrapper(*args, *kwargs):
+        if USER_RECOMMENDATION_SWITCH:
+            return func(*args, **kwargs)
+        else:
+            return {}, 200
```
Comment (aspadiyath) [inline app/utils.py:38]: Will throw an error in frontend the way this is written at the moment.

Will need to be:
```return {'message': 'success', 'recommendations': []}, 200```

**Review by aspadiyath — CHANGES_REQUESTED** [review]
I agree with roy on number 1, if we move on from the experiementing phase, we'll have to implement a system to check if the class should be on or off, rather than the whole system. Other than that, just one fix.

**Comment by aspadiyath** [comment]
> 2. Will this affect the dashboard too?
If we fix the code for my request, it should be fine.
> 3. How is the frontend handling this change? Are we showing nothing, or will the UI look broken? We may need to send back a special response code if the switch is off, so the frontend knows what to do
It just shows no posts, it looks fine.

File: app/utils.py (hunk 1 above)
Comment (tailorck) [reply app/utils.py:38]: noted.

**Review by aspadiyath — APPROVED** [review]

File: app/utils.py [hunk 2]
```diff
@@ -26,7 +27,19 @@ def wrapper(*args, **kwargs):
     wrapper.__name__ = func.__name__
     return wrapper
 
+  
+def global_user_recommendation_switch(func):
 
+    @wraps(func)
+    def wrapper(*args, *kwargs):
```
Comment (aspadiyath) [inline app/utils.py:34]: Should be **kwargs

File: app/utils.py [hunk 3]
```diff
@@ -26,7 +27,19 @@ def wrapper(*args, **kwargs):
     wrapper.__name__ = func.__name__
     return wrapper
 
+  
+def global_user_recommendation_switch(func):
 
+    @wraps(func)
```
Comment (aspadiyath) [inline app/utils.py:33]: module initialization error: name 'wraps' is not defined

**Review by aspadiyath — COMMENTED** [review]
Errors showing up in logs

**Pull request merged**